src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/StackGuard.cs

```diff
@@ -14,9 +14,14 @@ internal sealed class StackGuard
 
         private int _executionStackCount;
 
-
         public bool TryEnterOnCurrentStack()
         {
+#if NETCOREAPP
```

Comment:
We also build this project for netstandard2.1. We can also use `TryEnsureSufficientExecutionStack` there.

```suggestion
#if NETCOREAPP || NETSTANDARD2_1
```